sdk/core/core-amqp/test/context.spec.ts

```diff
@@ -35,6 +35,215 @@ describe("ConnectionContextBase", function() {
+    context.wasConnectionCloseCalled.should.equal(false);
+    context.connection.should.instanceOf(Connection);
+    context.connection.options.properties!.product.should.equal("MSJSClient");
+    context.connection.options.properties!["user-agent"].should.equal("/js-amqp-client");
+    context.connection.options.properties!.version.should.equal("1.0.0");
+    context.cbsSession.should.instanceOf(CbsClient);
+  });
+
+  it("should allow specifying a port", function() {
+    const connectionString =
+      "Endpoint=sb://hostname.servicebus.windows.net/;SharedAccessKeyName=sakName;SharedAccessKey=sak;EntityPath=ep";
+    const path = "mypath";
+    const config = ConnectionConfig.create(connectionString, path);
+    config.port = 1111;
+    const context = ConnectionContextBase.create({
+      config: config,
+      connectionProperties: {
+        product: "MSJSClient",
+        userAgent: "/js-amqp-client",
+        version: "1.0.0"
+      }
+    });
+    should.exist(context.config);
+    should.exist(context.connection);
+    should.exist(context.connectionId);
+    should.exist(context.connectionLock);
+    should.exist(context.negotiateClaimLock);
+    context.connection.options.port!.should.equal(1111);
+    context.wasConnectionCloseCalled.should.equal(false);
+    context.connection.should.instanceOf(Connection);
+    context.connection.options.properties!.product.should.equal("MSJSClient");
+    context.connection.options.properties!["user-agent"].should.equal("/js-amqp-client");
+    context.connection.options.properties!.version.should.equal("1.0.0");
+    context.cbsSession.should.instanceOf(CbsClient);
+  });
+
+  it("should have a default port (5671)", function() {
+    const connectionString =
+      "Endpoint=sb://hostname.servicebus.windows.net/;SharedAccessKeyName=sakName;SharedAccessKey=sak;EntityPath=ep";
+    const path = "mypath";
+    const config = ConnectionConfig.create(connectionString, path);
+    const context = ConnectionContextBase.create({
+      config: config,
+      connectionProperties: {
+        product: "MSJSClient",
+        userAgent: "/js-amqp-client",
+        version: "1.0.0"
+      }
+    });
+    should.exist(context.config);
+    should.exist(context.connection);
+    should.exist(context.connectionId);
+    should.exist(context.connectionLock);
+    should.exist(context.negotiateClaimLock);
+    context.connection.options.port!.should.equal(5671);
+    context.wasConnectionCloseCalled.should.equal(false);
+    context.connection.should.instanceOf(Connection);
+    context.connection.options.properties!.product.should.equal("MSJSClient");
+    context.connection.options.properties!["user-agent"].should.equal("/js-amqp-client");
+    context.connection.options.properties!.version.should.equal("1.0.0");
+    context.cbsSession.should.instanceOf(CbsClient);
+  });
+
+  it("should allow setting host and hostname to different values when using websockets", function() {
```

Comment:
Should there be a test for the 'default values when using websockets' like you have for the normal TCP version above?